lbz/configuration.py

```diff
@@ -0,0 +1,45 @@
+from abc import abstractmethod
+from os import getenv
+from typing import Any, Callable, Generic, Optional, TypeVar
+
+from lbz.aws_boto3 import SSM
+
+T = TypeVar("T")
+
+
+class BaseValue(Generic[T]):
+    def __init__(
+        self,
+        key: str,
+        parser: Optional[Callable[..., T]] = None,
+        default: T = None,
+    ):
+        self.key = key
+        self.parser = parser
+        self.default = default
+        self._value: T
+
+    @abstractmethod
+    def getter(self) -> Optional[T]:
+        pass
+
+    @property
+    def value(self) -> T:
+        if not hasattr(self, "_value"):
```

Comment:
I think I prefer to set `_value` equal to `None` in init instead of relying on this line and have the problems below 😉

Reply:
Then return Type becomes optional and we land were we started

Reply:
Fortunately, the newest version handled that correctly 👍